### PR TITLE
linux: Added functions to enable faster telemetry data retrieval.

### PR DESCRIPTION
**Pull request by lgdacunh**
Moved telemetry data area support detection into separate function. Added possibility to modify data transfer chunk size. Enable telemetry extraction up to specified data area.

### CODE REVIEW

File: src/nvme/linux.c [hunk 1]
```diff
@@ -122,17 +122,47 @@ int nvme_fw_download_seq(int fd, __u32 size, __u32 xfer, __u32 offset,
...
-static int nvme_get_telemetry_log(int fd, bool create, bool ctrl, bool rae,
-				  struct nvme_telemetry_log **buf, enum nvme_telemetry_da da,
-				  size_t *size)
+enum nvme_telemetry_da nvme_get_telemetry_max_da(int fd, size_t *max_data_tx)
+{
+	struct nvme_id_ctrl id_ctrl;
+	int err = nvme_identify_ctrl(fd, &id_ctrl);
+
+	if (err) {
+		perror("identify-ctrl");
```
Comment (igaw) [inline src/nvme/linux.c:131]: Unfortunately, we can't use printf & Co in the library.

File: src/nvme/linux.c [hunk 2]
```diff
@@ -122,17 +122,47 @@ int nvme_fw_download_seq(int fd, __u32 size, __u32 xfer, __u32 offset,
...
+		errno = EINVAL;
+		return 0;
```
Comment (igaw) [inline src/nvme/linux.c:133]: I think we should stay with the current pattern and just return the error status and have the a function argument for the result, as we have with the `max_data_tx` argument.

File: src/nvme/linux.c (hunk 1 above)
Comment (lgdacunh) [reply src/nvme/linux.c]: Removed!

File: src/nvme/linux.c (hunk 2 above)
Comment (lgdacunh) [reply src/nvme/linux.c]: Done as suggested!

File: src/nvme/linux.c [hunk 3]
```diff
@@ -207,6 +227,14 @@ static int nvme_get_telemetry_log(int fd, bool create, bool ctrl, bool rae,
 		goto free;
 	}
 
+	if (dalb == 0) {
+		fprintf(stderr, "ERROR: No telemetry data block");
```
[the remark this answers is not in the document]
Comment (lgdacunh) [reply src/nvme/linux.c]: Done as suggested!

**Comment by igaw** [comment]
Thanks!

**Pull request merged**